static/js/protein/chart.ts

```diff
@@ -58,11 +58,21 @@ export interface InteractionLink extends SimulationLinkDatum<ProteinNode> {
   score: number;
 }
 
+// Represents interaction graph as a list of nodes and list of links
 export interface InteractionGraphData {
   nodeData: ProteinNode[];
   linkData: InteractionLink[];
 }
 
+// Represents interaction graph depthwise as two parallel nested lists.  For each index i,
+// nodeDataNested[i] will contain the nodes added in i iterations of BFS,
+// and linkDataNested[i] will contain the links added in i iterations of BFS,
+// along with any links between nodesDataNested[i] and existing nodes in the graph.
+export interface InteractionGraphDataNested {
```

Comment:
Is `InteractionGraphDataNested` a good name for this? I was also considering `InteractionGraphDataDepthwise`.

Reply:
Got it that's a great name! And yeah `InteractionGraphData` is the input type to the draw function.

Reply:
Oh right BFS is just an implementation detail I'll remove it from the description.  And got it I'll collect all the types I use in a single file then.  

Unfortunately I'm doing BFS from the client right now.  I'll PR this soon for you guys to look at so we can decide what should be moved to flask.